astropy/io/fits/fitstime.py

```diff
@@ -241,21 +329,37 @@ def _convert_time_column(col, column_info):
         The time coordinate column to be converted to Time.
     column_info : dict
         Column-specific time reference frame override information.
+    global_info : dict
+        Global time reference frame information.
     """
 
     # The code might fail while attempting to read FITS files not written by astropy.
 
-    # Read in time coordinate column as Time
-    if col.shape[-1] == 2 and col.ndim > 1:
-        col = Time(col[..., 0], col[..., 1], format='jd',
-                   scale=column_info['scale'],
-                   location=column_info['location'])
-    else:
-        warnings.warn(
-            'Time column {} is not in the astropy required (jd1, jd2) format. '
-            'Hence, it will not be read as astropy Time'.format(col.info.name),
-            AstropyUserWarning)
-    return col
+    try:
+        # ISO-8601 is the only string representation of time in FITS
+        if col.info.dtype.kind == 'S':
+            return Time(col, format='fits', scale=column_info['scale'],
+                        location=column_info['location'])
+
+        # TODO: verify that unit is 'd' and jdref is 0.0
+        if col.shape[-1] == 2 and col.ndim > 1:
+            return Time(col[..., 0], col[..., 1], format='jd',
+                        scale=column_info['scale'],
+                        location=column_info['location'])
+
+        if column_info['format'] == 'gps':
+            return Time(col, format='gps', location=column_info['location'])
+
+        ref_time = Time(global_info['ref_time_value'], scale=column_info['scale'],
+                        format=global_info['ref_time_format'],
+                        location=column_info['location'])
+
+        scale = 'tai' if column_info['scale'] == 'utc' else column_info['scale']
+        delta_time = TimeDelta(col, format=column_info['format'], scale=scale)
+
+        return ref_time + delta_time
+    except:
```

Comment:
Need to catch error message and report the error back to the user via a warning.